client/src/components/DashboardLayout.tsx

```diff
@@ -0,0 +1,187 @@
+import { HamburgerMenuIcon, SunIcon } from '@radix-ui/react-icons';
+import { useTheme } from 'next-themes';
+import Image from 'next/image';
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import { config } from '../config';
+import { Box } from '../ui/Box';
+import { Button } from '../ui/Button';
+import { Container } from '../ui/Container';
+import { Dialog, DialogContent, DialogTrigger } from '../ui/Dialog';
+import { Flex } from '../ui/Flex';
+import { IconButton } from '../ui/IconButton';
+
+interface NavItemProps {
+  href: string;
+  children: React.ReactNode;
+}
+
+const NavItem = ({ href, children }: NavItemProps) => {
+  const router = useRouter();
+  return (
+    <Link href={href} passHref>
+      <Box
+        css={{
+          py: '$3',
+          px: '$3',
+          br: '$2',
+          alignSelf: 'start',
+          backgroundColor: router.pathname === href ? '$gray5' : 'transparent',
+
+          '&:hover': {
+            backgroundColor: router.pathname === href ? undefined : '$gray5',
+          },
+
+          '&:active': {
+            backgroundColor: '$gray5',
+          },
+        }}
+        target={href.includes('https') ? '_blank' : undefined}
+        rel="noreferrer"
+        as="a"
+      >
+        {children}
+      </Box>
+    </Link>
+  );
+};
+
+interface LayoutProps {
+  children?: React.ReactNode;
+}
+
+export const DashboardLayout = ({ children }: LayoutProps) => {
+  const { theme, setTheme, resolvedTheme } = useTheme();
+
+  let src;
+
+  switch (resolvedTheme) {
+    case 'light':
+      src = '/images/new_logo_stxstats_black.svg';
+      break;
+    case 'dark':
+      src = '/images/new_logo_stxstats_white.svg';
+      break;
+    default:
+      src =
+        'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7';
+      break;
+  }
+
+  return (
+    <>
+      <Dialog>
+        <DialogTrigger asChild>
+          <IconButton
+            css={{
+              alignSelf: 'start',
+              '@lg': {
+                display: 'none',
+              },
+            }}
+          >
+            <HamburgerMenuIcon />
+          </IconButton>
+        </DialogTrigger>
+        <DialogContent>
+          <Flex
+            css={{ flex: 1 }}
+            as={'nav'}
+            direction={'column'}
+            justify={'between'}
+            align={'start'}
+          >
+            <Box>
+              <Image height={38} width={81} src={src} alt="Stx stats logo" />
+              <Flex css={{ mt: '$10' }} direction={'column'} gap={'2'}>
+                <NavItem href={'/dashboard'}>Overview</NavItem>
```

Comment:
What do you think about removing the next link component from NavItem.
Something like:
```tsx
<Link href={href} passHref>
 <NavItem>Overview</NavItem>
</Link>
<Link href={config.twitterUrl} passHref>
 <NavItem rel="noreferrer"  target="_blank"> Twitter </NavItem>
</Link>
``` 

it's a bit more verbose but allow more customisation of the props. The NavItem component could accept an `active` prop that is adding some styling, and DashboardLayout will be responsible to check which one is active.